### PR TITLE
Add `system.user_processes` table

### LINKED ISSUE #50422: Add system.user_processes system table
**Use case**

Add table `system.user_processes`, which will show memory usage and profile counters per user. based on the info stored in [ProcessListForUser](https://github.com/ClickHouse/ClickHouse/blob/c0d6f9297d8d776cde434207795284d8eb5654ce/src/Interpreters/ProcessList.h#L253).

In theory, information can be get from `system.processes` grouping by `user`. Motivation for adding a separate table:
* Memory usage per user is not a sum of memory usage per query for the user. It may happen when memory is allocated in one scope and deallocated in another (which is usually a bug).
* It may be not convenient to sum profile events (it's a map)
* Entry for the user is stored even if there are no running queries. Info like peak memory usage may be useful.

### PR DESCRIPTION
**Pull request by antaljanosbenjamin**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `system.user_processes` table and `SHOW USER PROCESSES` query to show memory info and ProfileEvents on user level.

### Documentation entry for user-facing changes

- [X] Documentation is written (mandatory for new features)

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit 96fe4b5107611a627b7981fdac2afe9304660e48 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50492/96fe4b5107611a627b7981fdac2afe9304660e48/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟡 pending</td></tr>
</table>

File: src/Interpreters/InterpreterShowUserProcessesQuery.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+#include <Interpreters/IInterpreter.h>
+#include <Parsers/IAST_fwd.h>
+
+
+namespace DB
+{
+
+/** Return list of currently executing queries.
+TODO(antaljanosbenjamin)
```
Comment (serxa) [inline src/Interpreters/InterpreterShowUserProcessesQuery.h:11]: a small cleanup is required

File: docs/en/operations/system-tables/user_processes.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+slug: /en/operations/system-tables/user_processes
+---
+# user_processes
+
+This system table is used for implementing the `SHOW USER PROCESSES` query.
```
Comment (serxa) [inline docs/en/operations/system-tables/user_processes.md:6]: Do we really need this query type? 

Actually system table is accessible regardless of this query, so we have to write here something more generic about this table itself (e.g. "table with information about user processes ...") and just add "see also" link to the query doc entry

Doc entry for `SHOW USER PROCESSES` is also required if we need the query.

Comment (antaljanosbenjamin) [reply docs/en/operations/system-tables/user_processes.md]: I removed the query.

**Review by serxa — APPROVED** [review]

**Pull request merged**